### PR TITLE
✨amp-experiment: Base functionality for Demo

### PR DESCRIPTION
**Pull request by torch2424**
relates to #20225

This implements some base functionality for amp-experiment for an upcoming initial demo. This includes support for some of the `attributes` type mutations (style with `color` and `background-color`, src, and href), as well as `characterData` mutations 😄 

# Example

Screen Shot 2019-04-04 at 5 42 25 PM

### CODE REVIEW

File: extensions/amp-experiment/1.0/mutation-parser.js [hunk 1]
```diff
@@ -28,6 +30,34 @@ const MUTATION_TYPES = [
...
+
+    // Allow Background color
+    if (value.match(/^background-color:\s#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/g)) {
+      return true;
+    }
+
+    return false;
+  },
+  src: value => {
+    return value.match(/^https:\/\//g);
```
Comment (calebcordry) [inline extensions/amp-experiment/1.0/mutation-parser.js]: may want to use `assertHttpsUrl` here and below https://github.com/ampproject/amphtml/blob/f52c37a1005a5403f9d7c55f54d645826a50c313/src/url.js#L297

File: extensions/amp-experiment/1.0/mutation-parser.js [hunk 2]
```diff
@@ -28,6 +30,34 @@ const MUTATION_TYPES = [
...
+const SUPPORTED_ATTRIBUTES = {
+  style: value => {
+
+    // Do not allow Important or HTML Comments
+    if (value.match(/(!\s*important|<!--)/g)) {
+      return false;
+    }
+
+    // Allow Color
+    if (value.match(/^color:\s#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/g)) {
```
Comment (calebcordry) [inline extensions/amp-experiment/1.0/mutation-parser.js]: Should the space be optional here?

File: extensions/amp-experiment/1.0/mutation-parser.js [hunk 3]
```diff
@@ -28,6 +30,34 @@ const MUTATION_TYPES = [
...
+      return true;
+    }
+
+    // Allow Background color
+    if (value.match(/^background-color:\s#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/g)) {
```
Comment (calebcordry) [inline extensions/amp-experiment/1.0/mutation-parser.js]: Im fine with this for now, but I feel like eventually the property validation should be separated from the value validation.

Comment (torch2424) [reply extensions/amp-experiment/1.0/mutation-parser.js]: Totally agree, after our demo, we will probably be moving this to the Validator. If not, I added an note here: https://github.com/ampproject/amphtml/issues/21705 😄

**Comment by torch2424** [comment]
@calebcordry Made requested changes / replied to comments 😄

**Comment by calebcordry** [comment]
👍👊🎉

**Comment by jeffjose** [comment]
:muscle: mode!

**Comment by torch2424** [comment]
Thanks @calebcordry @jeffjose 😄 

Merging...

File: extensions/amp-experiment/1.0/mutation-parser.js (hunk 3 above)
Comment (honeybadgerdontcare) [reply extensions/amp-experiment/1.0/mutation-parser.js]: nit/bikeshedding:

Why did you go with just `\s` and not `\s*` after the colon (`:`)?

And why no whitespace before the colon or before/after the entire value set?

not nit-picking:

Why the `/g`? shouldn't there be only one `background-color`? what does the global accomplish or am I misunderstanding it (happens with regexs)

😁

**Pull request merged**